qcodes/instrument_drivers/Keysight/N9030B.py

```diff
@@ -0,0 +1,542 @@
+        start_offset = self.start_offset()
+
+        if abs(val - start_offset) >= 1:
+            self.log.warning(
+                f"Could not set start offset to {val} setting it to "
+                f"{start_offset}"
+            )
+        if val >= stop_offset or abs(val - stop_offset) < 10:
+            self.log.warning(f"Provided start frequency offset {val} Hz was "
+                             f"greater than preset stop frequency offset "
+                             f"{stop_offset} Hz. Provided start frequency "
+                             f"offset {val} Hz is set and new stop freq offset"
+                             f" is: {self.stop_offset()} Hz.")
+
+    def _set_stop_offset(self, val: float) -> None:
+        """
+        Sets stop offset for frequency in the plot
+        """
+        start_offset = self.start_offset()
+        self.write(f":SENSe:LPLot:FREQuency:OFFSet:STOP {val}")
+        stop_offset = self.stop_offset()
+
+        if abs(val - stop_offset) >= 1:
+            self.log.warning(
+                f"Could not set stop offset to {val} setting it to "
+                f"{stop_offset}"
+            )
+
+        if val <= start_offset or abs(val-start_offset) < 10:
+            self.log.warning(f"Provided stop frequency offset {val} Hz was "
+                             f"less than preset start frequency offset "
+                             f"{start_offset} Hz. Provided stop frequency "
+                             f"offset {val} Hz is set and new start freq offset"
+                             f" is: {self.start_offset()} Hz.")
+
+    def _get_data(self, trace_num: int) -> ParamRawDataType:
+        """
+        Gets data from the measurement.
+        """
+        self.cont_meas("OFF")
+        try:
+            data_str = self.ask(f":READ:{self.measurement}{trace_num}?")
+            data = np.array(data_str.rstrip()).astype("float64")
+        finally:
+            self.cont_meas("ON")
+
+        return data
+
+    def setup_log_plot_sweep(self,
+                             start_offset: float,
+                             stop_offset: float,
+                             npts: int
+                             ) -> None:
+        """
+        Sets up the Log Plot measurement sweep for Phase Noise Mode.
+        """
+        if "LPLot" in self._available_meas():
+            self.measurement("LPLot")
+        else:
+            raise RuntimeError("Log Plot measurement is not available on your "
+                               "Keysight N9030B instrument with Phase Noise "
+                               "mode.")
+
+        self.start_offset(start_offset)
+        self.stop_offset(stop_offset)
+        self.npts(npts)
+
+    def autotune(self) -> None:
+        """
+        Autotunes frequency
+        """
+        self.write(":SENSe:FREQuency:CARRier:SEARch")
+
+
+class N9030B(VisaInstrument):
+    """
+    Driver for Keysight N9030B PXA signal analyzer.
```

Comment:
Suggest adding more details to the doc string here about the instrument and what modes are included.